lib/spack/spack/ci/common.py

```diff
@@ -42,6 +43,35 @@
 # this exists purely for testing purposes
 _urlopen = web_util.urlopen
 
+#: Names of the core CI jobs, which are those not dependent on a specs
+core_job_names = ["cleanup", "copy", "noop", "reindex", "signing"]
+
+#: Names of all possible jobs, which include those involving specs
+all_job_names = ["any", "build", "test"] + core_job_names
+
+#: default job settings
+default_job_settings = {
+    "build-job": {
+        "script": ["cd {env_dir}", "spack env activate --without-view .", "spack ci rebuild"]
+    },
+    "noop-job": {"script": ['echo "All specs already up-to-date, nothing to rebuild."']},
+    "test-job": {
+        "script": [
+            "cd {env_dir}",
+            "spack env activate --without-view .",
+            "spack ci rebuild --tests",
+        ]
+    },
+}
+
+#: Override job settings
+override_job_settings = {
+    "any-job-remove": {"tags": SPACK_RESERVED_TAGS},
```

Comment:
This doesn't have to change here, but I think the reserved tags and signing job tags here should probably be left to configuration. The `"aws"` tag is definitely not portable outside of our setup.

Something like
```
ci:
    pipeline-gen:
    - build-jobs:
        protected:
            tags: [....] 
        public:
            tags: [....]
```